File: frontend/src/api/messageService.ts
import { legacyApi } from './axiosConfig';

export interface MessageDTO {
  id: string;
  senderId: string;
  senderName: string;
  receiverId: string;
  receiverName: string;
  content: string;
  timestamp: string;
  read: boolean;
}

const messageService = {
  getMyMessages: async (userId: string): Promise<MessageDTO[]> => {
    const response = await legacyApi.get<MessageDTO[]>(`/messages/user/${userId}`);
    return response.data;
  },

  sendMessage: async (senderId: string, receiverId: string, content: string): Promise<MessageDTO> => {
    const response = await legacyApi.post<MessageDTO>(
      `/messages?senderId=${senderId}&receiverId=${receiverId}`,
      content,
      { headers: { 'Content-Type': 'text/plain' } }
    );
    return response.data;
  },

  markAsRead: async (id: string): Promise<void> => {
    await legacyApi.patch(`/messages/${id}/read`);
  },
};

export default messageService;

File: backend/src/main/java/com/internbridge/backend/service/MessageService.java
package com.internbridge.backend.service;

import com.internbridge.backend.dto.response.MessageResponseDTO;
import java.util.List;
import java.util.UUID;

public interface MessageService {
    MessageResponseDTO sendMessage(UUID senderId, UUID receiverId, String content);
    List<MessageResponseDTO> getMessagesForUser(UUID userId);
    void markAsRead(UUID messageId);
}
